### PR TITLE
fix(autoware_behavior_path_planner_common): fix containerOutOfBounds warning

### PR DESCRIPTION
**Pull request by veqcc**
## Description

This is a fix based on cppcheck `containerOutOfBounds` error

```
planning/behavior_path_planner/autoware_behavior_path_planner_common/src/utils/path_utils.cpp:481:39: error: Out of bounds access in expression 'base_s.back()' because 'base_s' is empty. [containerOutOfBounds]
  for (double s = eps; s < base_s.back() - eps; s += resample_interval) {
                                      ^
planning/behavior_path_planner/autoware_behavior_path_planner_common/src/utils/path_utils.cpp:494:41: error: Out of bounds access in expression 'base_s.back()' because 'base_s' is empty. [containerOutOfBounds]
      end_pose, start_pose, (base_s.back() - new_s.at(i)) / base_s.back());
                                        ^
```

You know this is a false positive in cppcheck.
This PR is just to avoid cppcheck warnings.

## Related links

**Parent Issue:**

- Link

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: planning/behavior_path_planner/autoware_behavior_path_planner_common/src/utils/path_utils.cpp [hunk 1]
```diff
@@ -471,34 +471,36 @@ std::vector<Pose> interpolatePose(
...
-  const std::vector<double> base_y{start_pose.position.y, end_pose.position.y};
-  std::vector<double> new_s;
-
-  constexpr double eps = 0.3;  // prevent overlapping
-  for (double s = eps; s < base_s.back() - eps; s += resample_interval) {
-    new_s.push_back(s);
-  }
+  const double distance =
+    autoware::universe_utils::calcDistance2d(start_pose.position, end_pose.position)
+};
```
Comment (maxime-clem) [inline planning/behavior_path_planner/autoware_behavior_path_planner_common/src/utils/path_utils.cpp:476]: This code does not look correct to me :thinking: Maybe an extra `}` was introduced by mistake ?

Comment (veqcc) [reply planning/behavior_path_planner/autoware_behavior_path_planner_common/src/utils/path_utils.cpp:476]: Thank you for informing me!
That must be a mistake, I'll fix it soon

Comment (veqcc) [reply planning/behavior_path_planner/autoware_behavior_path_planner_common/src/utils/path_utils.cpp]: @maxime-clem 
I have fixed the problem. I am sorry to bother you, but could you review it again?

File: planning/behavior_path_planner/autoware_behavior_path_planner_common/src/utils/path_utils.cpp [hunk 2]
```diff
@@ -471,14 +471,15 @@ std::vector<Pose> interpolatePose(
 {
   std::vector<Pose> interpolated_poses{};  // output
 
-  const std::vector<double> base_s{
-    0, autoware::universe_utils::calcDistance2d(start_pose.position, end_pose.position)};
+  const double distance =
+    autoware::universe_utils::calcDistance2d(start_pose.position, end_pose.position);
+  const std::vector<double> base_s{0, distance};
```
Comment (maxime-clem) [inline planning/behavior_path_planner/autoware_behavior_path_planner_common/src/utils/path_utils.cpp:476]: I think using an `int` as the 1st argument here could cause the wrong constructor to be called (number 3 in https://en.cppreference.com/w/cpp/container/vector/vector).
```suggestion
  const std::vector<double> base_s{0.0, distance};
```
Sorry for the annoying comments :sweat_smile:

Comment (veqcc) [reply planning/behavior_path_planner/autoware_behavior_path_planner_common/src/utils/path_utils.cpp:476]: Agreed, I'll fix it right now.

**Review by maxime-clem — APPROVED** [review]

**Pull request merged**